statsmodels/tsa/vector_ar/var_model.py

```diff
@@ -1287,17 +1461,20 @@ def reorder(self, order):
 #-------------------------------------------------------------------------------
 # VAR Diagnostics: Granger-causality, whiteness of residuals, normality, etc.
 
-    def test_causality(self, equation, variables, kind='f', signif=0.05,
-                       verbose=True):
-        """Compute test statistic for null hypothesis of Granger-noncausality,
-        general function to test joint Granger-causality of multiple variables
+    def test_causality(self, causing, kind='f', signif=0.05, verbose=True):
```

Comment:
> The problem I have with the current situation is that the results in section 1 of the notebook I provided are not the same as those in section 2, because in the former there is one variable which is neither contained in causing nor in caused.

we need that, those test two different null hypothesis: With a left out variable, we test whether the causing variable has and additional granger causal effect.  It's a partial effect conditioning on other regressors, similar to a t-test or t-value for a single parameter.

Currently, we don't have a model for that yet, because the regressors are assumed to be the same in all `equations`, but we will need an extension where we can drop selectively irrelevant parameters or variables from just some equations. i.e. we need to be able to put zero constraints or other linear constraints on the full short-run parameter array.

I don't remember where I have seen it. Stata's vargranger only makes tests for all equations separately, i.e. all pairwise granger causality hypothesis conditional on other regressors and for `all` other variables in one equation, if I read the documentation correctly